graphx/src/main/scala/org/apache/spark/graphx/lib/ConnectedComponents.scala

```diff
@@ -29,13 +29,14 @@ object ConnectedComponents {
    *
    * @tparam VD the vertex attribute type (discarded in the computation)
    * @tparam ED the edge attribute type (preserved in the computation)
-   *
    * @param graph the graph for which to compute the connected components
-   *
+   * @param maxIterations the maximum number of iterations to run for
    * @return a graph with vertex attributes containing the smallest vertex in each
    *         connected component
    */
-  def run[VD: ClassTag, ED: ClassTag](graph: Graph[VD, ED]): Graph[VertexId, ED] = {
+  def run[VD: ClassTag, ED: ClassTag](graph: Graph[VD, ED],
+                                      maxIterations: Int = Int.MaxValue): Graph[VertexId, ED] = {
```

Comment:
for this one, we should also just overload it so we don't break binary compatibility.